site/source/tutorials/002_creating_an_application/index.md

```diff
@@ -28,58 +28,62 @@ You also need to be familiar with TypeScript as Dojo uses it extensively.
+{% instruction 'The `mount` operation in `main.ts`:' %}
+
+{% include_codefile 'demo/initial/biz-e-corp/src/main.ts' line:7 %}
 
 {% section %}
 
 ## Widgets
 
-Widgets are the basic building blocks of Dojo's user interface. They combine both the visual and behavioral aspects of a component into a single element. These aspects are encapsulated within the widget's implementation. The widget then exposes properties and methods that allow other components to interact with it. Consider the following diagrams:
+Up until now, we've return virtual DOM that represents HTML elements directly from the `renderer`, however that wouldn't scale very well as an application grow.
+
+That were widgets come into play, widgets are the basic building blocks of Dojo's user interface. They combine both the visual and behavioral aspects of a component into a single element. These aspects are encapsulated within the widget's implementation. The widget then exposes properties and methods that allow other components to interact with it. Consider the following diagrams:
 
 <img src="../resources/html_js.svg" title="HTML and JavaScript" class="half-width"/><img src="../resources/widget.svg" title="Widget" class="half-width"/>
 
 The first diagram shows a traditional HTML + JavaScript architecture. Since the visual (HTML) and behavioral (JavaScript) aspects of the application are publicly accessible, the application's components can be manipulated directly which can lead to the HTML and JavaScript getting out of sync with each other. Extensive test suites are often needed to make sure that this does not happen.
 
 The second image shows how widgets ensure that components only interact according to their design intent. The widget encapsulates its visual and behavioral aspects. It then exposes properties and methods that allow other components to interact with it. By providing a controlled interface, it is much easier to keep the visual and behavioral aspects of the widget synchronized.
 
-In our demo application, we only have one widget, the HelloWorld widget:
+In our demo application, we don't currently have any widgets, but let's abstract the virtual DOM from `main.ts` out into a dedicated widget. You'll notice that to render a widget, we use a different pragma, `w()`, than we used before to render nodes that directly represent the DOM nodes.
```

Comment:
we don't currently -> we do not currently

from `main.ts` out into -> from `main.ts` into

You'll -> You will